src/server/console/default_configuration.cpp

```diff
@@ -200,5 +200,13 @@ std::shared_ptr<mir::ConsoleServices> mir::DefaultServerConfiguration::the_conso
             BOOST_THROW_EXCEPTION((
                 std::runtime_error{
                     std::string{"Unknown console provider: "} + provider}));
-        });
+        };
+
+    // TODO: this is clearly not threadsafe, but then neither is CachedPtr
```

Comment:
That doesn't need to be called out here. *All* this "server configuration" code is intended to be used only during the single threaded initialization phase: prior to starting the server (and the associated threads).